### DISCUSSION TITLE
Overrides in Func

### DISCUSSION THREAD

**Discussion by Yeah69** [Yeah69/MrMeeseeks.DIE#15] [General]
Hi @dadhi,

I've made some adjustments to the overrides of Funcs. In my opinion, the overrides in DIE work very close to the overrides in Dry.Ioc, but not exactly the same. I can show you quickly with some tests I wrote.

Here how it works with Dry.Ioc:
https://github.com/Yeah69/DryIoc.Playground/blob/main/DryIoc.Playground/Case3.cs

Here the pendant of DIE:
https://github.com/Yeah69/MrMeeseeks.DIE/blob/main/Test/Func/ReductionCaseTransientScope.cs

The difference is how the re-iteration to `Parent` is handled, while Dry.Ioc would take over the override from the previous iteration, DIE "resets". See the last output: 23 for Dry.Ioc and 0 for DIE.
That is due to an optimization (which I call "function reduction") without which DIE would have an explosion of code generation. I've tested DIE on a hobby application that I once wrote:
https://github.com/Yeah69/BFF/blob/experimental/MrMeeseeks.DIE/Composition.DIE/Container.cs
Once I got the necessary container configuration figured out, DIE ran into an OutOfMemoryException right before generating. Even if my machine would have enough memory the generated code would have several millions of lines, which isn't practical even if it is "just generated". So I figured out the optimization an it is down to 4k lines of code & works good for the hobby app.

I think I'll stick with this slightly alternative version for now. What do you think?

**Comment by dadhi** [comment]
@Yeah69 That's the interesting case.

I would take it to my tests to play.

For now, I can the behavior of DryIoc because of passing `InputArgs` (Func arguments)  through the Request chain down to `Lazy<Parent>` dependency. DryIoc chains the service "request" even across the dynamic Resolve call dependencies like in case of `Lazy`.

I may probably put a stopper on argument propagation if I found the recursive service down the line, 
then go up via parent chain and see if the recursive service was outside/before the Func in parent hierarchy.

Let's see.

**Reply by dadhi** [reply]
Tracking it here https://github.com/dadhi/DryIoc/issues/514
